app/views/Project/Tagging/Export/utils.ts

```diff
@@ -0,0 +1,203 @@
+import { isDefined } from '@togglecorp/fujs';
+import _ts from '#ts';
+
+import { Matrix2dProperties } from '#types/newAnalyticalFramework';
+import {
+    Level,
+    ReportStructure,
+} from '#types';
+import { AnalysisFramework } from './types';
+
+export const SECTOR_FIRST = 'sectorFirst' as const;
+export const DIMENSION_FIRST = 'rowFirst' as const;
+
+export function getWidgets(framework: AnalysisFramework | undefined | null) {
+    if (!framework) {
+        return undefined;
+    }
+    const primaryWidgets = framework.primaryTagging?.map((v) => v.widgets)
+        .flat().filter(isDefined);
+    const secondaryWidgets = framework.secondaryTagging;
+    const allWidgets = [
+        ...(primaryWidgets || []),
+        ...(secondaryWidgets || []),
+    ];
+
+    return allWidgets;
+}
+
+function createId<T extends string | number>(...args: [T, T, ...T[]]) {
+    return args.join('-');
+}
+
+// NOTE: This function generates row first level
+const transformLevelsRowFirst = (
+    matrix2dProperties: Matrix2dProperties,
+    includeSubColumn: boolean,
+) => {
+    const rowFirstLevels = matrix2dProperties.rows.map((row) => {
+        const sublevels = row.subRows.map((subRow) => {
+            const columns = matrix2dProperties.columns.map((column) => (
+                {
+                    id: createId(column.clientId, row.clientId, subRow.clientId),
```

Comment:
Are we sure we have row and subRow clientIds, we refactored to use key inside widget properties so this could be "key"

Reply:
@tnagorra can we verify this? what should we use? right now only clientId's are present

Reply:
It should be "key" because we have already merged the "clientId" to "key" refactoring